splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DDLSingleTableConstantOperation.java

```diff
@@ -390,13 +390,7 @@ protected void dropIndex(TableDescriptor td, ConglomerateDescriptor conglomerate
         final long tableConglomId = td.getHeapConglomerateId();
         final long indexConglomId = conglomerateDescriptor.getConglomerateNumber();
         TxnView uTxn = userTxnManager.getRawTransaction().getActiveStateTxn();
-        //get the top-most transaction, that's the actual user transaction
         TransactionController tc = lcc.getTransactionExecute();
-        TxnView t = uTxn;
-        while(t.getTxnId()!= Txn.ROOT_TRANSACTION.getTxnId()){
-            uTxn = t;
-            t = uTxn.getParentTxnView();
-        }
```

Comment:
Hey @jyuanca , just curious, why was this causing the issue? Do you know what this code was trying to accomplish?

Reply:
I don't know why it uses root txn id to drop an index. With auto commit off, in the following DDL sequences, it has problem to see schema test2, because the root txn id is less than the txn that created schema test2

autocommit off;
create table test1.t1(c1 int);
create schema test2;
create table test2.tab2 (c1 int not null primary key, c2 int, c3 int);
create index test2.ti on test2.tab2(c1 desc);
drop index test2.ti;